assets/reader-activation/auth.js

```diff
@@ -54,299 +54,307 @@ const convertFormDataToObject = ( formData, includedFields = [] ) =>
-		googleLoginElements.forEach( googleLoginElement => {
-			const googleLoginForm = googleLoginElement.closest( 'form' );
+				messageContentElement.innerHTML = '';
+				form.style.opacity = 0.5;
+			};
 
-			const checkLoginStatus = metadata => {
-				fetch(
-					`/wp-json/newspack/v1/login/google/register?metadata=${ JSON.stringify( metadata ) }`
-				).then( res => {
-					res.json().then( ( { message, data } ) => {
-						if ( googleLoginForm.endLoginFlow ) {
-							googleLoginForm.endLoginFlow( message, res.status, data );
+			form.endLoginFlow = ( message, status, data, redirect ) => {
+				if ( message ) {
+					const messageNode = document.createElement( 'p' );
+					messageNode.textContent = message;
+					messageContentElement.appendChild( messageNode );
+				}
+				if ( status === 200 && data ) {
+					const authenticated = !! data?.authenticated;
+					readerActivation.setReaderEmail( data.email );
+					readerActivation.setAuthenticated( authenticated );
+					if ( authenticated ) {
+						if ( redirect ) {
+							window.location = redirect;
 						}
-					} );
+					} else {
+						form.replaceWith( messageContentElement.parentNode );
+					}
+				}
+				form.style.opacity = 1;
+				submitButtons.forEach( button => {
+					button.disabled = false;
 				} );
 			};
 
-			googleLoginElement.addEventListener( 'click', () => {
-				if ( googleLoginForm?.startLoginFlow ) {
-					googleLoginForm.startLoginFlow();
+			/**
+			 * Handle auth form submission.
+			 */
+			form.addEventListener( 'submit', function ( ev ) {
+				ev.preventDefault();
+				const body = new FormData( ev.target );
+				if ( ! body.has( 'email' ) || ! body.get( 'email' ) ) {
+					return;
 				}
+				readerActivation.setReaderEmail( body.get( 'email' ) );
+				form.startLoginFlow();
+				fetch( form.getAttribute( 'action' ) || window.location.pathname, {
+					method: 'POST',
+					headers: {
+						Accept: 'application/json',
+					},
+					body,
+				} )
+					.then( res => {
+						container.setAttribute( 'data-form-status', res.status );
+						res
+							.json()
+							.then( ( { message, data } ) => {
+								form.endLoginFlow( message, res.status, data, body.get( 'redirect' ) );
+							} )
+							.catch( () => {
+								form.endLoginFlow();
+							} );
+					} )
+					.catch( () => {
+						form.endLoginFlow();
+					} );
+			} );
 
-				const metadata = googleLoginForm
-					? convertFormDataToObject( new FormData( googleLoginForm ), [ 'lists[]' ] )
-					: {};
-				metadata.current_page_url = window.location.href;
-				const authWindow = window.open(
-					'about:blank',
-					'newspack_google_login',
-					'width=500,height=600'
-				);
-				fetch( '/wp-json/newspack/v1/login/google' )
-					.then( res => res.json().then( data => Promise.resolve( { data, status: res.status } ) ) )
-					.then( ( { data, status } ) => {
-						if ( status !== 200 ) {
-							if ( authWindow ) {
-								authWindow.close();
+			/**
+			 * Third party auth.
+			 */
```

Comment:
The third-party handling too does not rely on a specific container.

Reply:
Moved outside the loop in b08d5bc.